src/Features/CSharp/Portable/AddParameter/CSharpAddParameterCodeFixProvider.cs

```diff
@@ -39,5 +43,49 @@ protected override ImmutableArray<string> TooManyArgumentsDiagnosticIds
 
         protected override ImmutableArray<string> CannotConvertDiagnosticIds
             => GenerateConstructorDiagnosticIds.CannotConvertDiagnosticIds;
+
+        protected override async Task<bool> HandleLanguageSpecificNodesAsync(SyntaxNode initialNode, SyntaxNode node, CodeFixContext context)
+        {
+            if (node is ConstructorInitializerSyntax constructorInitializer)
```

Comment:
does VB work without any changes?

Reply:
I added tests for VB in 472f16c. In VB those invocations are handled by the *method invocation* part. In the `Me.New()` case the fix offers to fix the constructor with the diagnostic on it, which causes a recursive call to itself, which then get caught by the compiler. I special cased this for C# by removing the constructor with the diagnostic from the list of candidates [here](https://github.com/dotnet/roslyn/pull/29061/commits/54e7a1f5d3eb768d36af2acb4eedcc35df734f11#diff-2be993d9daf8f488e68fd76185093e8dR74).
The fix for this behavior in VB would be totally unrelated to what is needed in C#.

Reply:
@MaStr11 Gotcha.  I don't have a great solution.  Overall,i would think you could do:

After you determine the arguments and potential methods to update, call into a virtual helper to ask if it wants to filter anything out.  Pass in any data you need.  C# would do nothing, but VB would say: "oh, i'm generating a constructor, if i'm in that constructor, don't include my containing constructor".

Note: i think you already have this logic in the this/base() codepath for C#.  So you might want to make this a general helper, and just call it from both paths.

Does that make sense?